src/toil/job.py

```diff
@@ -1930,6 +1977,44 @@ def updateJobReqs(self, fileSize, actions):
             def isPopulated(self):
                 return self.__dict__ != {}
 
+        class HarbingerFile(object):
+            """
+            A class to handle the various harbinger file operations.
+            """
+            def __init__(self, outer, fileStoreID=None, cachedFileName=None):
+                """
+                Returns the harbinger file name for a cahed file, or for a job store ID
+                :param class outer: The 'self' object of the outer class
+                :param str fileStoreID:
+                :param str cachedFileName:
+                :return: Harbinger file name
+                :rtype: str
+                """
+                assert (fileStoreID and not cachedFileName) or (cachedFileName and not fileStoreID)
```

Comment:
`assert (fileStoreID is None) != (cachedFileName is None)` is shorter and more specific.

I generally don't like equating None and False-ish values. It's a source for bugs.